File: clientSide/lib/features/music/models/Song.dart
class Song {
  final String name;
  final String artist;
  final String image;
  final String url;

  Song({required this.name, required this.artist, required this.image,required this.url});

  factory Song.fromJson(Map<String, dynamic> json) {
    return Song(
      name: json['SongTitle'],
      artist: json['Artist'],
      image: json['Photo'],
      url: json['url']
    );
  }
}

File: serverSide/model/Song.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;
const Album = require('./Album'); 
const songSchema = new Schema({
    SongTitle: {
        type: String,
        required: true
    },
    Artist: {
        type: String,
        required: true
    },
    Photo: {
        type: String,
        required: true
    },
    Album: {
        type: Schema.Types.ObjectId,
        ref: 'Album',
        required: false
    },
    url: {
        type: String,
        required: true
    },
});

module.exports = mongoose.model('Song', songSchema);
